File: CS-IOS/Classes/SDWebImage/UIImageView+CSSDWebImage.m
//
// Created by Rene Dohan on 06/01/15.
// Copyright (c) 2015 creative_studio. All rights reserved.
//

@import UIKit;
@import SDWebImage;

#import "UIImageView+CSSDWebImage.h"
#import "CSLang.h"

@implementation UIImageView (CSSDWebImage)

- (UIImageView *)setImageNSURL:(NSURL *)url {
    return [self imageURL:url.absoluteString onSuccess:nil];
}

- (UIImageView *)imageURL:(NSString *)url onSuccess:(void (^)(UIImageView *imageView))onSuccess {
    __weak typeof(self) _self = self;
    self.sd_imageIndicator = SDWebImageProgressIndicator.barIndicator;
    [self sd_setImageWithURL:[NSURL URLWithString:url]
            placeholderImage:nil options:SDWebImageRetryFailed progress:nil
                   completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType, NSURL *imageURL) {
                       if (!error) runWith(onSuccess, _self);
                   }];
    return self;
}

- (UIImageView *)setImageURL:(NSString *)url {
    return [self imageURL:url onSuccess:nil];
}

@end

File: Renetik/Renetik/Classes/SDWebImage/UIImageView+CSSDWebImage.swift
//
// Created by Rene Dohan on 12/23/19.
//

import Foundation
import UIKit
import SDWebImage
import RenetikObjc

public extension UIImageView {

    @discardableResult
    public func image(url: URL, onSuccess: ((UIImageView) -> Void)? = nil) -> UIImageView {
        sd_imageIndicator = SDWebImageProgressIndicator.default
        sd_setImage(with: url, placeholderImage: nil, options: .retryFailed, progress: nil,
                completed: { image, error, cacheType, imageURL in error.isNil { onSuccess?(self) } })
        return self
    }

    @discardableResult
    public func image(url: String, onSuccess: ((UIImageView) -> Void)? = nil) -> UIImageView {
        URL(string: url).notNil { image(url: $0, onSuccess: onSuccess) }
                .elseDo { "Url for image was invalid: \(url)" }
        return self
    }
}